### PR TITLE
fix Issue 19319 - No line number when std.math is missing for x ^^ y

### PR DESCRIPTION
**Pull request by ibuclaw**
There's no gagging, and so loadStdMath will call fatal() if there's no std.math module available.

Leaving the code in, as gagging loadStdMath will be required if backend decides to do something with PowExp in the future.

### CODE REVIEW

**Comment by ibuclaw** [comment]
Huh, there's not even any line numbering, maybe this requires a little extra in order to add that then.

**Comment by ibuclaw** [comment]
I'll do something different later then.

File: src/dmd/expressionsem.d [hunk 1]
```diff
@@ -9462,19 +9462,28 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
...
+                }
+                result = exp;
                 return;
             }
-            result = exp;
-            return;
+        }
+        else
+        {
+            assert(mmath !is null);
```
Comment (WalterBright) [inline src/dmd/expressionsem.d]: `assert(0)` will do the job just fine, as:
```
if (cond) { ... } else { assert(!cond); }
```
seems faintly absurd.

Comment (ibuclaw) [reply src/dmd/expressionsem.d]: This is `version (none) { ... } else { assert(!cond); }`

**Comment by ibuclaw** [comment]
Fixed it up.  Removed e2ir override for PowExp, as it has been dead code for a long time.

File: src/dmd/expressionsem.d [hunk 2]
```diff
@@ -9464,17 +9468,8 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
...
-            // Leave handling of PowExp to the backend, or throw
-            // an error gracefully if no backend support exists.
-            if (Expression ex = typeCombine(exp, sc))
-            {
-                result = ex;
-                return;
-            }
-            result = exp;
-            return;
+            e.error("`%s` requires std.math for `^^` operators", e.toChars());
```
Comment (Geod24) [inline src/dmd/expressionsem.d]: Backtips around `std.math`

**Review by WalterBright — APPROVED** [review]

**Review by jacob-carlborg — APPROVED** [review]

**Pull request merged**